### PR TITLE
8257563: Remove excessive include of klass.inline.hpp

### PR DESCRIPTION
**Pull request by iklam**
klass.inline.hpp is unnecessarily included by two popular header files: oop.inline.hpp and modRefBarrierSet.inline.hpp

These two `#include` lines recursively pull in many header files. Removing them reduce the total number of header inclusion for building HotSpot from 260096 to 254773, or about 2%.

Notes to reviewer. The main changes are in the header files: 

* src/hotspot/share/classfile/systemDictionaryShared.hpp
* src/hotspot/share/gc/shared/modRefBarrierSet.inline.hpp
* src/hotspot/share/oops/instanceKlass.hpp
* src/hotspot/share/oops/instanceKlass.inline.hpp
* src/hotspot/share/oops/oop.inline.hpp

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8257563](https://bugs.openjdk.java.net/browse/JDK-8257563): Remove excessive include of klass.inline.hpp

### Reviewers
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to 96ce2128cfc43b512326557397683e468650b673
 * [Thomas Stuefe](https://openjdk.java.net/census#stuefe) (@tstuefe - **Reviewer**) ⚠️ Review applies to 96ce2128cfc43b512326557397683e468650b673
 * [Stefan Karlsson](https://openjdk.java.net/census#stefank) (@stefank - **Reviewer**) ⚠️ Review applies to 96ce2128cfc43b512326557397683e468650b673

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1552/head:pull/1552`
`$ git checkout pull/1552`

### CODE REVIEW

**Review by dholmes-ora — APPROVED** [review]
Seems okay.

Thanks,
David

**Review by tstuefe — APPROVED** [review]
LGTM. Thank you for doing this. My slow notebook thanks you.

..Thomas

**Review by stefank — APPROVED** [review]
Nice! One comment:

File: src/hotspot/cpu/s390/s390.ad [hunk 1]
```diff
@@ -1325,6 +1325,8 @@ source_hpp %{
 // To keep related declarations/definitions/uses close together,
 // we switch between source %{ }% and source_hpp %{ }% freely as needed.
 
+#include "oops/klass.inline.hpp"
+
```
[the remark this answers is not in the document]
Comment (iklam) [reply src/hotspot/cpu/s390/s390.ad:1329]: We have #includes in the .ad files today. I am not sure what the rules are. I am guessing it is "add includes that are specific to the C code inside this .ad file". So I'd like to keep my changes in s390.ad as is.

```
$ find . -name \*.ad | xargs grep -l '^#include '
./cpu/aarch64/aarch64.ad
./cpu/aarch64/gc/z/z_aarch64.ad
./cpu/aarch64/gc/shenandoah/shenandoah_aarch64.ad
./cpu/ppc/ppc.ad
./cpu/x86/gc/z/z_x86_64.ad
./cpu/x86/gc/shenandoah/shenandoah_x86_64.ad
./cpu/x86/gc/shenandoah/shenandoah_x86_32.ad
./cpu/x86/x86.ad
```

[the remark this answers is not in the document]
Comment (stefank) [reply src/hotspot/cpu/s390/s390.ad:1329]: OK. So platform-specific files seems to have includes. Got it. Thanks.

**Comment by iklam** [comment]
Thanks @tstuefe @stefank @dholmes-ora for the review. I've tested tier1,builds-tier2,builds-tier3,builds-tier4,builds-tier5 on mach5, plus locally building s390x,ppc64le.
/integrate

**Pull request closed without merging**